qvm-tools/qubes-hcl-report

```diff
@@ -17,10 +17,11 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library; if not, see <https://www.gnu.org/licenses/>.
 
+set -euo pipefail
```

Comment:
Adding `set -e` here means you should review every command in the script and consider if it failing should really be fatal. In a script like this, I'd prefer to have some field missing if it wasn't detected than having no output at all (which is the case right now). Because of that, IMO better to _not_ use `set -e`.

The cases I found already:
 - network devices (you may have none on PCI)
 - same for USB (although unlikely)
 - SCSI (this I'm actually surprised worked for you, do you really have some SCSI disk?)
 - HAP and Interrupt Remapping - when those are missing, it should report "no", instead of crashing the script
 - same for VT-x, VT-d

There are probably more.